lib/MusicBrainz/Server/Data/Search.pm

```diff
@@ -748,12 +748,20 @@ sub external_search
         my $dismax = $adv ? 'false' : 'true';
         $search_url_string = "http://%s/ws/2/%s/?query=%s&offset=%s&max=%s&fmt=jsonnew&dismax=$dismax&web=1";
     } else {
-        if ($adv) {
-            $search_url_string = "http://%s/%s/edismax?q=%s&start=%s&rows=%s&wt=mbjson&fl=score";
-        } else {
-            $search_url_string = "http://%s/%s/select?q=%s&start=%s&rows=%s&wt=mbjson&fl=score&defType=dismax";
+        my $endpoint = "advanced";
+        if (!$adv)
+        {
+            # Solr has a bug where the dismax end point behaves differently
+            # from edismax when the query size is 1. This is a fix for that
+            # See https://issues.apache.org/jira/browse/SOLR-12409
+            if (split(/[\P{Word}_]+/, $query) == 1) {
```

Comment:
I suggest adding `, 2` as the last argument to `split` so that it stops right after it finds the first split (since you only care that it finds one, not how many).

Reply:
Done.